client/engine/engine.go

```diff
@@ -88,6 +88,7 @@ func New(msg messageservice.MessageService, chain chainservice.ChainService, sto
 
 	e.store = store
 
+	e.fromLedger = make(chan consensus_channel.Proposal, 100)
```

Comment:
I'm kind of amazed that anything worked without this line -- I guess the bug is combining with the other problem so we didn't see a `panic` (which would be what I would have expected). Is that your understanding too?

Reply:
Just re-read your PR description. It's not a panic but an infinite wait. Still bad.

Reply:
> Just re-read your PR description. It's not a panic but an infinite wait. Still bad.

In general I think nil channels can be a huge footgun; it can be really easy to miss the infinite wait if that wait is happening in some go-routine we're not paying attention to.

Reply:
Yikes. We need a linting rule for this or something